src/pass.jl

```diff
@@ -60,7 +60,6 @@ macro pass(transform)
         import Cassette.__overdub_generator__
         struct $Pass <: $Cassette.AbstractPass end
         (::Type{$Pass})(ctxtype, reflection) = $transform(ctxtype, reflection)
-        Core.eval($__module__, $Cassette.overdub_definition($line, $file))
```

Comment:
Sorry, just getting round to this now. 

The mechanism removed by this PR exists to avoid the classic generate function world age issue:

```
julia> struct Foo end

julia> (::Type{Foo})(::Type{T}) where {T} = :(string("foo", $T))

julia> @generated g(t, x) = t(x)
g (generic function with 1 method)

julia> struct Bar end

julia> (::Type{Bar})(::Type{T}) where {T} = :(string("foo", $T))

julia> g(Bar(), 1)
ERROR: MethodError: no method matching Bar(::Type{Int64})
The applicable method may be too new: running in world age 25568, while current world is 25570.
Closest candidates are:
  Bar(::Type{T}) where T at REPL[5]:1 (method too new to be called from this world context.)
  Bar() at REPL[4]:1 (method too new to be called from this world context.)
Stacktrace:
 [1] #s1#3(::Any, ::Any, ::Any) at ./REPL[3]:1
 [2] (::Core.GeneratedFunctionStub)(::Any, ::Vararg{Any,N} where N) at ./boot.jl:522
 [3] top-level scope at none:0
```

I'm not sure how to cleanly work around it otherwise...

I'm having trouble reproducing the method overwrite warning on Julia 1.2.0-DEV.454 (commit e949f89b39) and Cassette master. Starting up with `julia --warn-overwrite=yes`, is there another flag I should use to repro?

Reply:
It does it without **any** flags for me.
But I've not tried on julia 1.2

Reply:
I have been seeing this generally, but I think you need to put it in a module that is precompiled.

Reply:
Hmm. Not seeing it there either, just loading:

```julia
module MyPkg

using Cassette

end # module
```

At the REPL for Julia 1.1 and Julia 1.2.0-DEV.454:

```jl
julia> using MyPkg
[ Info: Precompiling MyPkg [bb27c774-b52d-11e8-00fa-c947777c4ef9]

julia>
```

Reply:
Try loading MagneticReadHead

Reply:
```

(v1.1) pkg> activate MyPkg

julia> using MyPkg
[ Info: Recompiling stale cache file /home/vchuravy/.julia/compiled/v1.1/MyPkg/hoTst.ji for MyPkg [2a6e8cfc-56e7-11e9-17f6-3b8c2c81ea6d]
WARNING: Method definition overdub(Cassette.Context{N, M, T, P, B, H} where H<:Union{Cassette.DisableHooks, Nothing} where B<:Union{Nothing, Base.IdDict{Module, Base.Dict{Symbol, Cassette.BindingMeta}}} where P<:Cassette.AbstractPass where T<:Union{Nothing, Cassette.Tag{N, X, E} where E where X where N<:Cassette.AbstractContextName} where M where N<:Cassette.AbstractContextName, Any...) in module Cassette at /home/vchuravy/.julia/packages/Cassette/xggAf/src/overdub.jl:508 overwritten in module MyPkg at /home/vchuravy/.julia/packages/Cassette/xggAf/src/overdub.jl:508.
WARNING: Method definition recurse(Cassette.Context{N, M, T, P, B, H} where H<:Union{Cassette.DisableHooks, Nothing} where B<:Union{Nothing, Base.IdDict{Module, Base.Dict{Symbol, Cassette.BindingMeta}}} where P<:Cassette.AbstractPass where T<:Union{Nothing, Cassette.Tag{N, X, E} where E where X where N<:Cassette.AbstractContextName} where M where N<:Cassette.AbstractContextName, Any...) in module Cassette at /home/vchuravy/.julia/packages/Cassette/xggAf/src/overdub.jl:521 overwritten in module MyPkg at /home/vchuravy/.julia/packages/Cassette/xggAf/src/overdub.jl:521.
```

`cat MyPkg/src/MyPkg.jl`:
```
module MyPkg

using Cassette

Cassette.@context Ctx
const mypass = Cassette.@pass (ctx, ref) -> ref.code_info

end # module
```